### DISCUSSION TITLE
Kick.com 160p quality

### DISCUSSION THREAD

**Discussion by abradila** [KevinBytesTheDust/CVAmp#147] [General]
Hi guys !

I realised that when the bot was watching a kick.com channel, the quality was 720p or 1080p which takes a lot of bandwidth.

I managed to modify the file sites.py, so the bot can watch every channel in 160p.

Here's my code if you want to put it on the next update, or if it can help people who uses the bot for kick.com : 

--------

class Kick(Instance):
    name = "KICK"

    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)

    def todo_every_loop(self):
        self.page.keyboard.press("Tab")

    def update_status(self):
        pass

    def todo_after_spawn(self):
        self.goto_with_retry(self.target_url)
        self.page.wait_for_timeout(1000)
        self.page.click("button:has-text('Quality')")
        self.page.click("li:has-text('160p')")
      
        
        if 'cloudflare' in self.page.content().lower():
            raise utils.CloudflareBlockException("Blocked by Cloudflare.")

**Comment by jlplenio** [comment]
Beautiful work @abradila. That was my plan all along with this modular approach! Do you know how to make a pull request?

**Reply by abradila** [reply]
nope ^^

**Reply by jlplenio** [reply]
Are you interested in learning it, i.e. officially contributing your code to this repository, or should I implement your code?

**Comment by l3franc3** [comment]
Thats great stuff!

I'd suggest to add a try/except on the
`self.page.click("li:has-text('160p')")`

Because some live streams only have de 720p option and instance crashes
